core/src/main/java/brooklyn/entity/rebind/ActivePartialRebindIteration.java

```diff
@@ -0,0 +1,158 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package brooklyn.entity.rebind;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import brooklyn.basic.BrooklynObject;
+import brooklyn.basic.BrooklynObjectInternal;
+import brooklyn.entity.Entity;
+import brooklyn.entity.basic.EntityInternal;
+import brooklyn.entity.rebind.persister.BrooklynMementoPersisterToObjectStore;
+import brooklyn.entity.rebind.persister.PersistenceActivityMetrics;
+import brooklyn.entity.rebind.transformer.CompoundTransformer;
+import brooklyn.management.ha.ManagementNodeState;
+import brooklyn.mementos.BrooklynMementoPersister;
+import brooklyn.mementos.BrooklynMementoRawData;
+import brooklyn.mementos.BrooklynMementoRawData.Builder;
+import brooklyn.mementos.Memento;
+import brooklyn.util.collections.MutableList;
+import brooklyn.util.collections.MutableSet;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * Pauses a set of existing entities, writes their state, applies a transformation, then reads them back.
+ */
+public class ActivePartialRebindIteration extends RebindIteration {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ActivePartialRebindIteration.class);
+    
+    protected Iterator<BrooklynObject> objectsToRebindInitial;
+    protected Collection<BrooklynObject> objectsToRebindFinal;
+    protected List<CompoundTransformer> transformers = MutableList.of();
+    
+    public ActivePartialRebindIteration(RebindManagerImpl rebindManager, 
+            ManagementNodeState mode,
+            ClassLoader classLoader, RebindExceptionHandler exceptionHandler,
+            Semaphore rebindActive, AtomicInteger readOnlyRebindCount, PersistenceActivityMetrics rebindMetrics, BrooklynMementoPersister persistenceStoreAccess
+            ) {
+        super(rebindManager, mode, classLoader, exceptionHandler, rebindActive, readOnlyRebindCount, rebindMetrics, persistenceStoreAccess);
+    }
+
+    @Override
+    protected boolean isRebindingActiveAgain() {
+        return true;
+    }
+    
+    public void setObjectIterator(Iterator<BrooklynObject> objectsToRebind) {
+        this.objectsToRebindInitial = objectsToRebind;
+    }
+    
+    public void applyTransformer(CompoundTransformer transformer) {
+        if (transformer!=null)
+            transformers.add(transformer);
+    }
+    
+    @Override
+    protected void doRun() throws Exception {
+        checkEnteringPhase(1);
+        Preconditions.checkState(readOnlyRebindCount.get()==Integer.MIN_VALUE, "Partial rebind in read-only mode not supported");
```

Comment:
This seems a strange ay to check if the state is read-only mode. I wouldn't have guess it from looking at the variable name.